lib/llvm-ast.ts

```diff
@@ -62,43 +81,44 @@ export class LlvmAstParser {
             if (beginEnd.length === 2) {
                 const begin = this.parsePoint(beginEnd[0], lastLineNo);
                 const end = this.parsePoint(beginEnd[1], begin.line);
-                return {type: this.locTypes.SPAN, begin, end};
+                return {type: LocTypes.SPAN, begin, end};
             } else {
-                return {type: this.locTypes.POINT, loc: this.parsePoint(span, lastLineNo)};
+                return {type: LocTypes.POINT, loc: this.parsePoint(span, lastLineNo)};
             }
         }
-        return {type: this.locTypes.NONE};
+        return {type: LocTypes.NONE};
     }
 
     // Link the AST lines with spans of source locations (lines+columns)
-    parseAndSetSourceLines(astDump) {
-        let lfrom = {line: null, loc: null},
-            lto = {line: null, loc: null};
+    parseAndSetSourceLines(astDump: ResultLine[]) {
+        let lfrom: any = {line: null, loc: null},
+            lto: any = {line: null, loc: null};
         for (const line of astDump) {
             const span = this.parseSpan(line.text, lfrom.line);
             switch (span.type) {
-                case this.locTypes.NONE: {
+                case LocTypes.NONE: {
                     break;
                 }
-                case this.locTypes.POINT: {
+                case LocTypes.POINT: {
                     lfrom = span.loc;
                     lto = span.loc;
                     break;
                 }
-                case this.locTypes.SPAN: {
+                case LocTypes.SPAN: {
                     lfrom = span.begin;
                     lto = span.end;
                     break;
                 }
             }
-            if (span.type !== this.locTypes.NONE) {
-                line.source = {from: lfrom, to: lto};
+            if (span.type !== LocTypes.NONE) {
+                // TODO: ResultLineSource doesn't have to/from
+                (line.source as any) = {from: lfrom, to: lto};
```

Comment:
Not sure if this is wrong or if `ResultLineSource` is wrong

Reply:
oh hmm that's weird...another typing mystery.